mrp_unbuild_rebuild_variant/models/mrp_unbuild_rebuild_variant.py

```diff
@@ -0,0 +1,276 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import _, api, exceptions, fields, models
+
+
+class MrpUnbuildRebuildVariant(models.Model):
+
+    _name = "mrp.unbuild.rebuild.variant"
+    _description = "Wrapper aroud the unbuild - rebuild process."
+
+    name = fields.Char(
+        default=lambda self: self.env["ir.sequence"].next_by_code(self._name),
+        readonly=True,
+    )
+    state = fields.Selection(
+        [("draft", "Draft"), ("done", "Done")],
+        required=True,
+        default="draft",
+        readonly=True,
+    )
+    stock_location_id = fields.Many2one("stock.location")
+    # TODO: WHAT SHOULD BE COMPUTED HERE
+    unbuild_product_id = fields.Many2one("product.product")
+    unbuild_bom_id = fields.Many2one("mrp.bom", compute="_compute_unbuild_product_id")
+    # if unbuild product is tracked:
+    unbuild_product_tracking = fields.Selection(
+        string="Unbuild Product Tracking", related="unbuild_product_id.tracking"
+    )
+    unbuild_lot_id = fields.Many2one(
+        "stock.production.lot",
+        domain="[('product_id', '=', unbuild_product_id)]",
+        required=True
+    )
+    # if unbuild product's bom includes tracked components:
+    unbuild_original_mo_id = fields.Many2one("mrp.production")
+    # if unbuild product's bom does not include tracked components:
+    quantity = fields.Float(default=1.0)
```

Comment:
Add a constraint, must be 1.0 if unbuild product is tracked, can be higher if product is not tracked.